ledger/participant-integration-api/src/main/resources/db/migration/postgres-appendonly/V100__Append_only_schema.sql

```diff
@@ -0,0 +1,753 @@
+);
+INSERT INTO temp_divulged_contracts (
+    event_sequential_id,
+    contract_id,
+    divulgees
+)
+SELECT
+    nextval('temp_divulgence_sequential_id') + ledger_end_sequential_id,
+    contract_id,
+    array_agg(contract_witness)
+FROM
+    participant_contract_witnesses
+        INNER JOIN participant_contracts USING (contract_id)
+        LEFT JOIN participant_events USING (contract_id),
+    temp_ledger_end_sequential_info
+WHERE (NOT create_stakeholders @> array[contract_witness])
+    AND exercise_consuming IS NULL -- create events only
+    AND (tree_event_witnesses IS NULL OR NOT tree_event_witnesses @> array[contract_witness])
+GROUP BY contract_id, ledger_end_sequential_id;
+
+INSERT INTO participant_events_divulgence (
+    event_sequential_id,
+    event_offset,
+    command_id,
+    workflow_id,
+    application_id,
+    submitters,
+    contract_id,
+    template_id,
+    tree_event_witnesses,
+    create_argument,
+    create_argument_compression
+)
+SELECT
+    event_sequential_id,
+    -- The following 5 fields are metadata of the transaction that lead to the divulgence.
+    -- We can't reconstruct this information from the old schema.
+    NULL, -- TODO append-only: this is the only reason why the event_offset is nullable. Use some non-null offset instead?
+    NULL,
+    'flyway-migration',
+    NULL,
+    NULL,
+    contract_id,
+    template_id,
+    divulgees,
+    create_argument,
+    create_argument_compression
+FROM temp_divulged_contracts INNER JOIN participant_contracts USING (contract_id);
+
+-- Drop temporary tables
+DROP TABLE temp_divulged_contracts;
+DROP TABLE temp_ledger_end_sequential_info;
+DROP SEQUENCE temp_divulgence_sequential_id;
+
+
+---------------------------------------------------------------------------------------------------
+-- Drop old tables, at this point all data has been copied to the new tables
+---------------------------------------------------------------------------------------------------
+
+DROP TABLE participant_contracts CASCADE;
+DROP TABLE participant_contract_witnesses CASCADE;
+DROP TABLE participant_events CASCADE;
+
+
+---------------------------------------------------------------------------------------------------
+-- Events table: view of all events
+---------------------------------------------------------------------------------------------------
+
+-- This view is used to drive the transaction and transaction tree streams,
+-- which will in the future also contain divulgence events.
+-- The event_kind field defines the type of event (numbers allocated to leave some space for future additions):
+--    0: divulgence event
+--   10: create event
+--   20: consuming exercise event
+--   25: non-consuming exercise event
+-- TODO append-only: EITHER only include columns that are used in queries that use this view OR verify that the query planning
+-- is not negatively affected by a long list of columns that are never used.
+CREATE VIEW participant_events
```

Comment:
This way all existing queries keep working. We can (and probably should) optimize some of the queries to use `select from participant_events_divulgence` instead of `select from participant_events where event_kind = 0` (and similar for other partitions).